yaml_cpp_vendor-extras.cmake.in

```diff
@@ -16,3 +16,6 @@ set(yaml_cpp_vendor_LIBRARIES ${YAML_CPP_LIBRARIES})
 set(yaml_cpp_vendor_INCLUDE_DIRS ${YAML_CPP_INCLUDE_DIR})
 
 list(APPEND yaml_cpp_vendor_TARGETS yaml-cpp)
+if(WIN32)
+  set_target_properties(${YAML_CPP_LIBRARIES} PROPERTIES INTERFACE_COMPILE_DEFINITIONS YAML_CPP_DLL)
+endif()
```

Comment:
I think it is worthwhile to add a comment to explain a bit about what is going on here from our investigation.  Something like:

```suggestion
# On Windows, yaml-cpp 0.7.0 requires that downstream consumers set YAML_CPP_DLL and not yaml_cpp_EXPORT in order to set dllimport properly.
# This behavior will likely change in future versions of yaml-cpp.
if(WIN32)
  set_target_properties(${YAML_CPP_LIBRARIES} PROPERTIES INTERFACE_COMPILE_DEFINITIONS YAML_CPP_DLL)
endif()
```

Reply:
Added [1370b67](https://github.com/ros2/yaml_cpp_vendor/pull/38/commits/1370b679a18b186f3c72bae335561939e1983afc)